4717_Sprint/9-Haoran/Demo.java

```diff
@@ -0,0 +1,61 @@
+import java.util.*;
+import java.time.*;
+import java.time.format.*;
+
+public class Demo {
+
+  private void siriMode(String input, Siri siri){
+
+    if (input.contains("sassy") && "normal".equals(siri.getChatbotMode())) {
+      siri.setChatbotMode("sassy");
+      System.out.println("SassySiri model on.");
+    } else if ("sassy".equals(siri.getChatbotMode()) && input.contains("normal") ) {
+      siri.setChatbotMode("normal");
+      System.out.println("Normal model on.");
+    }else throw new IllegalArgumentException("Siri model does not exist.");
+  }
+
+  private void response(String input, Siri siri){
+    switch (siri.getChatbotMode()) {
+      case "sassy":
+        Siri sassySiri = new SassySiri();
+        if (input.contains("what time")) {
+          sassySiri.outputTime();
+        } else {
+          sassySiri.response();
+        }
+        break;
+
+      case "normal":
+        if (input.contains("what time")) {
+          siri.outputTime();
+        } else {
+          siri.response();
+        }
+        break;
+    }
```

Comment:
Why are you using a switch instead of a nested if statement? Seeing as you only have to options and omit the default action.